### ISSUE TITLE
[Fleet]: No data available under Data Streams for installed Agents.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#654]
**Kibana version:**  8.4 Snapshot Kibana cloud environment

**Host OS and Browser version:** All, All

**Build details:**
```
VERSION: 8.4.0 Snapshot cloud
BUILD: 53518
COMMIT: 8a54c809495bc08782359073d9392f25c8eb6499
```

**Preconditions:**
1. 8.4 Snapshot Kibana cloud environment should be available.
2. Agent should be installed using Policy. 

**Steps to reproduce:**
1. Navigate to Fleet>Agents tab.
2. Observe agents installed Healthy under Agents tab.
3. Navigate to Data Streams and observe no data available for any integrations.

**Expected Result:**
Data should be available under Data Streams for installed Agents.

**Logs:**
[elastic-agent-diagnostics-2022-06-29T05-32-50Z-00.zip](https://github.com/elastic/kibana/files/9008082/elastic-agent-diagnostics-2022-06-29T05-32-50Z-00.zip)

**Screenshots:**

![9](https://user-images.githubusercontent.com/77374876/176373874-08b5f927-b083-4679-96f0-bbff7c8f6da5.JPG)
![10](https://user-images.githubusercontent.com/77374876/176373878-b2ed9881-621e-4a68-bced-23bc65dc11a9.JPG)

**Comment by jen-huang** [comment]
cc @pierrehilbert

**Comment by pierrehilbert** [comment]
@amolnater-qasource : As it seems to be the same for each OS, could you please provide me linux agent logs?

**Comment by amolnater-qasource** [comment]
Hi @jen-huang @pierrehilbert 
We have revalidated on latest 8.4 Snapshot Kibana cloud-staging environment and found that the issue is **still reproducible**.
- No data available under Data Streams for installed Agents.

**Build details:**
BUILD: 54160
COMMIT: b509d2466e88bf6c4386d8dd5fe89b5c8a54a97f

**Logs:**
**Windows:**
[Windows.zip](https://github.com/elastic/elastic-agent/files/9037694/Windows.zip)

**Linux:**
[Linux.zip](https://github.com/elastic/elastic-agent/files/9037695/Linux.zip)

**MAC:**
[MAC.zip](https://github.com/elastic/elastic-agent/files/9037698/MAC.zip)

Please let us know if anything else is required from our end.
Thanks

**Comment by pierrehilbert** [comment]
In addition, could you please share with me the link to the kibana?

**Comment by amolnater-qasource** [comment]
Hi @pierrehilbert 
We have shared the required 8.4 Snapshot Kibana details over slack.

Please let us know if anything else is required from our end.
Thanks

**Comment by jen-huang** [comment]
Thanks for the investigation @pierrehilbert! Is the lack of `event.ingested` field something to look into?

**Comment by amolnater-qasource** [comment]
Hi @kpollich 
We have revalidated this issue on latest 8.4 Snapshot and found it **fixed** now.
- Data is available under Data Streams for installed Agents.

**Build details:**
BUILD: 54370
COMMIT: 27befe47a084f7b046426aa3edac01293d6e407b

**Screenshot:**
![11](https://user-images.githubusercontent.com/77374876/178247237-1c30d4a3-6f20-4db4-b55e-e5a5685d5df0.JPG)

Hence marking this as **QA:Validated**.
Thanks

**Comment by amolnater-qasource** [comment]
Hi @pierrehilbert 
While testing logstash output on 8.4 Snapshot Kibana cloud environment, we have observed this issue reproducible.
<details><summary>Steps followed:</summary>
<p>

1. Create 3 certs ca, client and logstash with below commands:

```
elasticsearch-certutil ca --pem
elasticsearch-certutil cert --name logstash --ca-cert C:\elk\elasticsearch\ca\ca.crt --ca-key C:\elk\elasticsearch\ca\ca.key --pem
elasticsearch-certutil cert --name client --ca-cert C:\elk\elasticsearch\ca\ca.crt --ca-key C:\elk\elasticsearch\ca\ca.key --pem

```
2. Convert the logstash key to pkcs8 as it is the only format supported by logstash:

`openssl pkcs8 -inform PEM -in logstash.key -topk8 -nocrypt -outform PEM -out logstash.pkcs8.key
`


3. Create logstash output-`My Logstash Output`(could be any):

Add logstash hosts: 10.10.10.10:5044 (machine ip)

Add below certs:

In Server SSL certificate we added- ca certificate.
In Client SSL certificate we added- client certificate.
In Client SSL certificate SSL key we added- client certificate key.

4. Added required configuration in pipelines.yml and created elastic-agent-pipeline.conf.

pipelines.yml:

```
- pipeline.id: elastic-agent-pipeline
  path.config: "C:\elk\logstash\config\elastic-agent-pipeline.conf"

```
**elastic-agent-pipeline.conf:**

```
input {
  elastic_agent {
    port => 5044
    ssl => true
    ssl_certificate_authorities => ["C:\elk\elasticsearch\ca\ca.crt"]
    ssl_certificate => "C:\elk\elasticsearch\logstash\logstash.crt"
    ssl_key => "C:\elk\elasticsearch\logstash\logstash.pkcs8.key"
    ssl_verify_mode => "force_peer"
  }
}

output {
  elasticsearch {
    hosts => ["https://10.10.10.10:9200"](https://10.10.10.10:9200/)
    api_key => "zorYA4ABQH2oZOWhPX_Q:_Y00OErrQGCRr5tJBE1_zQ"
    data_stream => true
    ssl => true
   # cacert => 
  }
}

```

5. Run logstash from bin using below command:

` logstash -f  C:\elk\logstash\config\elastic-agent-pipeline.conf `

6. Create new policy with integrations output selected as- `My Logstash Output`.
7. Run Agent install command for second agent.
8. Observe data for System integration for this installed agent under Data Streams tab.
9. Add OSQuerybeat integration and observe data for OSQuerybeat data too.

</p>
</details>

**After some time Kibana get logged out itself and we observed all data streams removed under Data Streams tab.**
However, we have observed new data under Discover tab.

**Screenshots:**
![15](https://user-images.githubusercontent.com/77374876/178462268-87d7ef44-56b8-4697-a333-3234f25664ed.png)
![14](https://user-images.githubusercontent.com/77374876/178462283-d23d52e6-7576-4db8-93f5-85ccdea1ba06.png)
![14](https://user-images.githubusercontent.com/77374876/178462396-418d5d90-080d-48a3-a42d-0f4c756b0ff6.JPG)

**Build details:**
```
VERSION: 8.4.0
BUILD: 54378
COMMIT: 17a2bcc82856ea7720f684a5f0e2ab0664517401
```

Hence we are **re-opening** this issue and sharing the kibana build over slack.

Thanks

**Comment by joshdover** [comment]
> **After some time Kibana get logged out itself and we observed all data streams removed under Data Streams tab.**
> However, we have observed new data under Discover tab.

Want to make sure I understand what's happening here. @amolnater-qasource you were able to see the data streams created from data ingested via Logstash, but then they disappeared after some time even though no ingest changes were made?

Could you check the `event.ingested` field on the documents you're seeing in discover and share a screenshot of the data including the `data_stream.dataset` and `data_stream.type` fields?

**Comment by amolnater-qasource** [comment]
Hi @joshdover 
Thank you for looking into this.

>you were able to see the data streams created from data ingested via Logstash, but then they disappeared after some time even though no ingest changes were made?

Yes we were able see the data streams, however after adding both the integrations (System & OSQueryManager) and passing few minutes we observed authentication error and also error pop-ups like shown below:
![image](https://user-images.githubusercontent.com/77374876/178718585-e94e4691-0429-412c-8045-3fd543b6fbdb.png)

After few failed attempts of logging in to environment because of session errors and we were able to login to the environment.
However when we navigated to Data Streams tab, none of the previous data streams were available.

Further please find below `event.ingested` field results under Discover tab:
![2](https://user-images.githubusercontent.com/77374876/178719739-e7b3b173-4588-43bc-b00a-f5578a9d56c1.JPG)
![3](https://user-images.githubusercontent.com/77374876/178719749-adaac0a3-aa58-4952-8949-52f045522c01.JPG)

**Json file for elastic_agent.metricbeat dataset:**
[event-ingested.txt](https://github.com/elastic/elastic-agent/files/9101497/event-ingested.txt)

Please let us know if anything else is required from our end.
Thanks

**Comment by joshdover** [comment]
Thanks for the information. I think the best course of action would be to have a developer dig into this on the cluster you have (if you haven't deleted it of course). @kpollich would you agree?

**Comment by kpollich** [comment]
Yes it'd be great to get access to this cluster to take a look at the data streams API response in detail.

**Comment by amolnater-qasource** [comment]
Hi @kpollich @joshdover  
We have shared the existing environment over slack.

Further we will be revalidating the shared issue under https://github.com/elastic/elastic-agent/issues/654#issuecomment-1181561902 on latest 8.4 Snapshot Kibana environment and will be logging a separate issue if found reproducible again.

Please let us know if anything else is required from our end.
Thanks

**Comment by amolnater-qasource** [comment]
Hi Team
We have revalidated this issue on latest 8.4 Snapshot and found it **reproducible there**.
- For reproducing this issue we need to run a live query at OSQuery tab and data under data streams gets removed.

We have logged a separate issue for this at https://github.com/elastic/elastic-agent/issues/721
Hence we are **closing** this.

Thanks
